### PR TITLE
commit for part 4

### PR DESCRIPTION
**Pull request by ashcreek**

### CODE REVIEW

File: src/part_04/1_questions [hunk 1]
```diff
@@ -1,49 +1,97 @@
...
+String[] cats = new String[10];
 
 3) What is a multidimensional array?
+It's a way of storing data in rows and columns instead of just in rows like a one dimensional array.
 
 4) What is the code required to create (instantiate) a 2-dimensional int array that is 6 high and 6 wide named "scores"?
+int[][] scores = int[6][6];
 
 5) Using code, how do you retrieve a String that is the 10th item in a traditional array named dogs?
+return dogs[10];
```
Comment (T-Davis) [inline src/part_04/1_questions]: Close but not quite

File: src/part_04/1_questions [hunk 2]
```diff
@@ -1,49 +1,97 @@
...
+This means that your code has tried to access an array farther than the array actually exists.
 
 7) Using one line of code, demonstrate how to retrieve the second to the last item from a int array.
+return myIntArray[myIntArray.length-2];
 
 8) What is the maximum allowable size of an array in Java?
+Integer.MAX_VALUE-5
 
 9) What is a bubble sort?
+It's a way of sorting through an array.
```
Comment (T-Davis) [inline src/part_04/1_questions]: The bubble sort is clearly a way to sort things. What else?

File: src/part_04/1_questions [hunk 3]
```diff
@@ -1,49 +1,97 @@
...
 
 16) What is a stack?
+It is also a stored set of information that you can add or remove from. However, it is first in- last out. Meaning you can only
+add or remove from the end.
 
 17) What differentiates stacks and queues?
+Stacks are added and removed from only the end. Queues are added from the end but removed from the beginning.
 
 18) When would you use an enhanced for loop?
+When you don't want to have to specify the length of what you are iterating through.
```
Comment (T-Davis) [inline src/part_04/1_questions]: Also when you don't need to know the index of the item

File: src/part_04/1_questions [hunk 4]
```diff
@@ -1,49 +1,97 @@
...
 
 21) What is a String?
+It is an object (under the String Class) but they behave like primitive variables (somewhat).
 
 22) What are a few common methods for operating on Strings?
+    String string = new String("The String")
+1. string.length();
+2. string.concat("is a good string")
+3. string.charAt(0);
+4. string.valueOf("The");
```
Comment (T-Davis) [inline src/part_04/1_questions]: valueOf() returns the string representation of the argument, so you wouldn't use a String, but you could use an int, Integer, boolean, char or others

File: src/part_04/1_questions (hunk 3 above)
Comment (ashcreek) [reply src/part_04/1_questions]: Ya, that makes sense!

File: src/part_04/1_questions (hunk 4 above)
Comment (ashcreek) [reply src/part_04/1_questions]: okay!

File: src/part_04/1_questions (hunk 1 above)
Comment (ashcreek) [reply src/part_04/1_questions]: return dogs[9];

**Pull request merged**